Src/Newtonsoft.Json/Linq/JToken.cs

```diff
@@ -2113,60 +2113,47 @@ public static JToken ReadFrom(JsonReader reader, JsonLoadSettings settings)
 
             if (reader.TokenType == JsonToken.None)
             {
-                if (!reader.Read())
+                if (!(settings != null && settings.CommentHandling == CommentHandling.Ignore ? reader.ReadAndSkipComments() : reader.Read()))
```

Comment:
Yeah, it was removing the duplication the throw that pushed it past the threshold for me, but also because I have a low threshold for these changes figuring it's better to discuss every such change I could see and reject some than second-guess too early.

I'm also inclined to give a bit more leeway when the side-effects of both are similar and when they're "consuming" side-effects.

Maybe line-splitting it would bring a bit more visual focus to the to the two expressions picked between, and slightly reduce the downside?